libevmasm/SemanticInformation.h

```diff
@@ -56,6 +56,10 @@ struct SemanticInformation
 	/// without altering the semantics. This means it cannot depend on storage or memory,
 	/// cannot have any side-effects, but it can depend on a call-constant state of the blockchain.
 	static bool movable(Instruction _instruction);
+	/// @returns true for instructions EXTCODEHASH, EXTCODESIZE and SLOAD. For the rest, same as
```

Comment:
Can you start with the generic condition first and then list examples last?

Also generic condition should be something like:

```
true if the instruction can be moved across instructions that do not modify storage or blockchain state (they can modify memory, though).
```

Reply:
Please also add: It also allows instruction to be moved across control-flow branches.